run_cart.sh

```diff
@@ -0,0 +1,147 @@
+        echo "CMD:"  
+        echo orterun --map-by node --timeout $timeout --mca mtl ^psm2,ofi -x FI_PSM2_DISCONNECT=1 --np $1 --hostfile $2 --output-filename testLogs/$4/sep_$test_$sep -x D_LOG_FILE=testLogs/$4/sep_$test_$sep/srv_output.log -x D_LOG_MASK=ERR -x CRT_PHY_ADDR_STR=ofi+psm2 -x OFI_INTERFACE=$interface -x CRT_CTX_SHARE_ADDR=$sep -x CRT_CTX_NUM=16 -x PATH -x LD_LIBRARY_PATH $3
+
+        echo "" 
+
+        orterun --map-by node --timeout $timeout --mca mtl ^psm2,ofi -x FI_PSM2_DISCONNECT=1 --np $1 --hostfile $2 --output-filename testLogs/$4/sep_$test_$sep -x D_LOG_FILE=testLogs/$4/sep_$test_$sep/srv_output.log -x D_LOG_MASK=ERR -x CRT_PHY_ADDR_STR=ofi+psm2 -x OFI_INTERFACE=$interface -x CRT_CTX_SHARE_ADDR=$sep -x CRT_CTX_NUM=16 -x PATH -x LD_LIBRARY_PATH $3 &
+
+        wait 5
+}
+
+function run_fg {
+        echo "CMD:"
+        echo orterun --map-by node --timeout $timeout --mca mtl ^psm2,ofi -x FI_PSM2_DISCONNECT=1 --np $1 --hostfile $2 --output-filename testLogs/$4/sep_$test_$sep -x D_LOG_FILE=testLogs/$4/sep_$test_$sep/srv_output.log -x D_LOG_MASK=ERR -x CRT_PHY_ADDR_STR=ofi+psm2 -x OFI_INTERFACE=$interface -x CRT_CTX_SHARE_ADDR=$sep -x CRT_CTX_NUM=16 -x PATH -x LD_LIBRARY_PATH $3
+
+        echo ""
+
+        orterun --map-by node --timeout $timeout --mca mtl ^psm2,ofi -x FI_PSM2_DISCONNECT=1 --np $1 --hostfile $2 --output-filename testLogs/$4/sep_$test_$sep -x D_LOG_FILE=testLogs/$4/sep_$test_$sep/srv_output.log -x D_LOG_MASK=ERR -x CRT_PHY_ADDR_STR=ofi+psm2 -x OFI_INTERFACE=$interface -x CRT_CTX_SHARE_ADDR=$sep -x CRT_CTX_NUM=16 -x PATH -x LD_LIBRARY_PATH $3
+
+        ret=$?
+        wait 5
+        if [ $ret != 0 ]; then
+                echo TEST FAILED!!! ret = $ret
+        else
+                echo TEST PASSED
+        fi
+        echo ""
+}
+
+function run_st {
+        echo "CMD:"  
+        echo orterun --map-by node --timeout $timeout --mca mtl ^psm2,ofi -x FI_PSM2_DISCONNECT=1 --np $1 --hostfile $2 --output-filename testLogs/$4/sep_$test_$sep -x D_LOG_FILE=testLogs/$4/sep_$test_$sep/srv_output.log -x D_LOG_MASK=ERR -x CRT_PHY_ADDR_STR=ofi+psm2 -x OFI_INTERFACE=$interface -x CRT_CTX_SHARE_ADDR=$sep -x CRT_CTX_NUM=16 -x PATH -x LD_LIBRARY_PATH $3 --message-sizes "b1048576,b1048576 0,0 b1048576,i2048,i2048 0,0 i2048,0"
+
+        echo "" 
+
+        orterun --map-by node --timeout $timeout --mca mtl ^psm2,ofi -x FI_PSM2_DISCONNECT=1 --np $1 --hostfile $2 --output-filename testLogs/$4/sep_$test_$sep -x D_LOG_FILE=testLogs/$4/sep_$test_$sep/srv_output.log -x D_LOG_MASK=ERR -x CRT_PHY_ADDR_STR=ofi+psm2 -x OFI_INTERFACE=$interface -x CRT_CTX_SHARE_ADDR=$sep -x CRT_CTX_NUM=16 -x PATH -x LD_LIBRARY_PATH $3 --message-sizes "b1048576,b1048576 0,0 b1048576,i2048,i2048 0,0 i2048,0"
+
+        ret=$?
+        wait 5
+        if [ $ret != 0 ]; then
+                echo TEST FAILED!!! ret = $ret
+        else
+                echo TEST PASSED
+        fi
+        echo ""
+}
+
+#Create server/client hostfile.
+prepare(){
```

Comment:
May be better this to move to common script instead of duplicate in each run?

Reply:
Sure.